src/events.rs

```diff
@@ -158,10 +163,21 @@ impl MuxedEvents {
     /// match against the one contained in each `notify::Event` received.
     /// Otherwise returns `Error` for a given registration failure.
     pub async fn add_file(&mut self, path: impl Into<PathBuf>) -> io::Result<PathBuf> {
-        self._add_file(path)
+        self._add_file(path, false)
     }
 
-    fn _add_file(&mut self, path: impl Into<PathBuf>) -> io::Result<PathBuf> {
+    /// Adds a given file to the event watch, allowing for files which do not
+    /// yet exist. Once the file is added, an event is immediately created for
+    /// the file to trigger reading it as soon as events are being read.
+    ///
+    /// Returns the canonicalized version of the path originally supplied, to
+    /// match against the one contained in each `notify::Event` received.
+    /// Otherwise returns `Error` for a given registration failure.
+    pub async fn add_file_initial_event(&mut self, path: impl Into<PathBuf>) -> io::Result<PathBuf> {
```

Comment:
Hmm, yeah, synthesizing a `Create` event from `MuxedEvents` is probably the best way to support in this feature without having to significantly rework how `MuxedLines` does things.

That said, can we make this function exposed only to the crate? Then if `MuxedLines::add_file_from_start` ever gets reworked such that `MuxedLines` can read from start on its own, this can be safely removed without breaking the API.

Reply:
Sorry for the slow updates here - was so glad to have our project done and I managed to use my fork to do it. I'll take a look at this again this week :)